### PR TITLE
[53] [Backend] As a logged in user, I can submit survey questions

### LINKED ISSUE #53: [Backend] As a logged in user, I can submit survey questions
## Why

- The user must be able to submit the answer for survey questions
 
## Acceptance Criteria

- Make a request with post method to API endpoint `/api/v1/responses` with body:
```
{
    "survey_id": "survey_id",
    "questions": [
        {
            "id": "question_id",
            "answers": [
                {
                    "id": "answer_id"
                }
            ]
        },
        {
            "id": "question_id",
            "answers": [
                {
                    "id": "answer_id",
                    "answer": "My answer"
                }
            ]
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by chiiida**
Resolves https://github.com/llleyelll/ic-surveys-ios/issues/53

## What happened 👀

- Implement submit survey request.
 
## Insight 📝

- The endpoint `/api/v1/responses` has a blank string response. It needs to handle and return as `EmptyResponse` instead.
Screen Shot 2021-07-08 at 11 51 42 AM

 
## Proof Of Work 📹

On success:
Screen Shot 2021-07-08 at 11 32 06 AM

On failure:
Screen Shot 2021-07-08 at 11 32 43 AM

### CODE REVIEW

**Review by blyscuit — APPROVED** [review]
Looks good, please check indents.

File: SurveyApp/Services/Network/API/APIResponseSerializer.swift [hunk 1]
```diff
@@ -43,6 +40,9 @@ final class APIResponseSerializer<T: Decodable>: ResponseSerializer {
             if response.statusCode > 300 {
                 let result = try errorSerializer.serialize(request: request, response: response, data: data, error: nil)
                 return .failure(result)
+            } else if data == " ".data(using: .utf8) {
+                let result = try successSerializer.serialize(request: request, response: response, data: emptyData, error: nil)
+                return .success(result)
             } else {
                 let result = try successSerializer.serialize(request: request, response: response, data: data, error: nil)
                 return .success(result)
```
Comment (markgravity) [inline SurveyApp/Services/Network/API/APIResponseSerializer.swift]: Let make it more readable by removing `else if` statement 👍 
```suggestion
            if response.statusCode > 300 {
                let result = try errorSerializer.serialize(request: request, response: response, data: data, error: nil)
                return .failure(result)
            }
            
            if data == " ".data(using: .utf8) {
                let result = try successSerializer.serialize(request: request, response: response, data: emptyData, error: nil)
                return .success(result)
            }
            
            let result = try successSerializer.serialize(request: request, response: response, data: data, error: nil)
            return .success(result)
```

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: SurveyApp/Services/Network/API/APIResponseSerializer.swift (hunk 1 above)
Comment (chiiida) [reply SurveyApp/Services/Network/API/APIResponseSerializer.swift]: I updated the code. Thank you for pointing that out 🎯 🙏

**Pull request merged**